tsfc/fem.py

```diff
@@ -41,57 +40,56 @@ def _tabulate(ufl_element, order, points):
     :arg ufl_element: element to tabulate
     :arg order: FIAT gives all derivatives up to this order
     :arg points: points to tabulate the element on
+    :arg entity: particular entity to tabulate on
     """
     element = create_element(ufl_element)
     phi = element.space_dimension()
     C = ufl_element.reference_value_size()
     q = len(points)
-    for D, fiat_table in iteritems(element.tabulate(order, points)):
-        reordered_table = fiat_table.reshape(phi, C, q).transpose(1, 2, 0)  # (C, q, phi)
-        for c, table in enumerate(reordered_table):
-            yield c, D, table
-
-
-def tabulate(ufl_element, order, points, epsilon):
-    """Same as the above, but also applies FFC rounding with
-    threshold epsilon and recognises cellwise constantness.
-    Cellwise constantness is determined symbolically, but we
-    also check the numerics to be safe."""
-    for c, D, table in _tabulate(ufl_element, order, points):
-        # Copied from FFC (ffc/quadrature/quadratureutils.py)
-        table[abs(table) < epsilon] = 0
-        table[abs(table - 1.0) < epsilon] = 1.0
-        table[abs(table + 1.0) < epsilon] = -1.0
-        table[abs(table - 0.5) < epsilon] = 0.5
-        table[abs(table + 0.5) < epsilon] = -0.5
+    for D, fiat_table in iteritems(element.tabulate(order, points, entity)):
+        if isinstance(fiat_table, Exception):
+            # In the case an exception is found in the fiat table, do not
+            # perform any rounding
+            gem_fail = gem.Failure((q, phi), fiat_table)
```

Comment:
OK, so we just use this to raise an error later, and the `(q, phi)` is something of the right shape.